src/errors.rs

```diff
@@ -22,23 +26,31 @@ pub enum ErrorKind {
     InvalidType(String),
 }
 
-#[derive(Debug)]
-/// Error struct
-pub struct EnvmntError {
-    /// Holds the error information
-    pub kind: ErrorKind,
-}
-
 impl Display for EnvmntError {
     /// Formats the value using the given formatter.
     fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        match self.kind {
-            ErrorKind::ReadFile(ref message, ref cause) => {
+        match self {
+            Self::ReadFile(ref message, ref cause) => {
                 writeln!(formatter, "{}", message)?;
                 cause.fmt(formatter)
             }
-            ErrorKind::Missing(ref msg) => writeln!(formatter, "{}", msg),
-            ErrorKind::InvalidType(ref msg) => writeln!(formatter, "{}", msg),
+            Self::Missing(ref msg) => writeln!(formatter, "{}", msg),
+            Self::InvalidType(ref msg) => writeln!(formatter, "{}", msg),
+        }
+    }
+}
+
+impl Error for EnvmntError
+{
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match self
+        {
+            // FsIOError does not properly implement std::error::Error, no chaining possible
```

Comment:
actually fsio is mine as well, you want maybe to push a PR there to sort that out there first and that way we can chain it?

Reply:
Not using it, but I'll have a look. It might me interesting for me.

Reply:
Ok, opened PR. Will open a separate PR for this here once merged.